system/src/system_cloud.cpp

```diff
@@ -38,8 +42,37 @@
 #include "deviceid_hal.h"
 #include "system_mode.h"
 
+#if PLATFORM_ID != PLATFORM_SPARK_CORE
+  #include "spark_wiring_timer.h"
+#endif // not PLATFORM_SPARK_CORE
+
 extern void (*random_seed_from_cloud_handler)(unsigned int);
 
+namespace {
+
+using namespace particle::system;
+
+#if PLATFORM_ID == PLATFORM_SPARK_CORE
+  particle::NullTimer _vitals_timer;
+  VitalsPublisher<std::function<int(ProtocolFacade *, int, void *)>, particle::NullTimer> _vitals(spark_protocol_post_description, _vitals_timer);
+#else // not PLATFORM_SPARK_CORE
+  std::function<void(void)> _periodic_callback = []() -> void {
```

Comment:
`&sp`

Reply:
`spark_protocol_instance()`

Reply:
The rest of this file is using `sp` an externally defined global from `system_cloud_internal.h`, which in turn sets it equal to `spark_protocol_instance()` in it's implementation.

Probably best to follow the pattern utilized in rest of the file. If the value of the `sp` abstraction is changed, then the behavior of the lambda will follow that of the file.